### PR TITLE
Fix access policy write validation

### PR DESCRIPTION
**Pull request by yayashuxue**
## Summary
- reject invalid access enum values and malformed allowlists on write paths instead of silently normalizing them
- keep legacy/corrupt read fallback in normalizeAccess, but require strict values for PATCH /api/doc/access and upload meta.access
- validate upload access before writing R2 document bytes, so malformed policy fails closed without partial doc writes

## Verification
- node test/access.test.js
- node test/remote-access-route.test.js
- node test/run.js
- mutation check: removing visibility validation makes access.test.js fail

### CODE REVIEW

**Review by serenakeyitan — COMMENTED** [review]
Post-merge review found one P1 access-policy regression. The existing 22-suite run is green, but the new upload test checks source ordering rather than executing this branch; the failing behavior is also still present on current main. Agent note: this reply was prepared and posted by breeze running locally for the active account.

---
_sent via [breeze](https://github.com/serenakeyitan/breeze) on behalf of @serenakeyitan_

File: worker/worker.js [hunk 1]
```diff
@@ -2277,6 +2322,27 @@ export default {
...
+        incoming = (meta && typeof meta === 'object') ? { ...meta } : {};
+        let prev = null;
+        try {
+          const prevRaw = await env.META.get(`meta:${slug}`);
+          if (prevRaw) prev = JSON.parse(prevRaw);
+        } catch {}
+        if (!incoming.access && prev && prev.access) {
+          incoming.access = prev.access;
+        }
+        if (incoming.access) {
```
Comment (serenakeyitan) [inline worker/worker.js:2338]: [P1] JSON-present falsy access values bypass strict validation. Both preservation and validation use truthiness here. For a new slug, `meta.access` set to `null`, `false`, `0`, or `""` skips `validateAccessWrite`, returns HTTP 200, writes R2, persists malformed metadata, and `accessFromMeta` then applies the legacy public/public fallback. This violates the PR’s fail-closed write contract. Please distinguish property absence from a present falsy value: inherit the previous policy only when `access` is absent, and validate whenever it is present. Add a route-level `fetch` test with `{ meta: { access: null } }` that asserts HTTP 400 plus zero `DOCS.put`/`META.put` calls. Agent note: this reply was prepared and posted by breeze running locally for the active account.

---
_sent via [breeze](https://github.com/serenakeyitan/breeze) on behalf of @serenakeyitan_
